x-pack/libbeat/reader/parquet/parquet.go

```diff
@@ -0,0 +1,99 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package parquet
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/apache/arrow/go/v11/arrow/memory"
+	"github.com/apache/arrow/go/v11/parquet"
+	"github.com/apache/arrow/go/v11/parquet/file"
+	"github.com/apache/arrow/go/v11/parquet/pqarrow"
+)
+
+// StreamReader parses parquet inputs from io streams.
+type StreamReader struct {
+	cfg          *Config
+	fileReader   *file.Reader
+	recordReader pqarrow.RecordReader
+}
+
+// NewStreamReader creates a new reader that can decode parquet data from an io.Reader.
+// It will return an error if the parquet file cannot be read.
+// NewStreamReader will read the entire contents of the file into memory, so very large files
+// can cause memory bottleneck issues.
+func NewStreamReader(r io.Reader, cfg *Config) (*StreamReader, error) {
```

Comment:
`NewStreamReader` gives a notion of reading of data in a streaming from the io.Reader `r` but it is not. It's read at once and held up it in memory.

Don't you think it should be named something else?